qcodes/dataset/sqlite/queries.py

```diff
@@ -829,10 +831,47 @@ def get_guid_from_run_id(conn: ConnectionPlus, run_id: int) -> str:
     Args:
         conn: database connection
         run_id: id of the run
+
+    Returns:
+        The guid of the run_id.
     """
     return select_one_where(conn, "runs", "guid", "run_id", run_id)
 
 
+def get_guid_from_multiple_run_ids(db_path: Union[str, Path],
+                                   run_ids: Iterable[int]) \
+                                   -> List[str]:
+    """
+    Expanding the functionality of get_guid_from_run_id to retrive guids
+    from database and for multiple run ids.
+
+    Args:
+        db_path: The path to the database file.
+        run_ids: An integer iterable of run ids to get their guids.
+
+    Returns:
+        A list of guids belonging to the inserted run_ids.
+    """
+
+    guids: List[str] = []
+
+    try:
+        conn = connect(db_path)
+        for run_id in run_ids:
+            if run_exists(conn=conn, run_id=run_id):
+                run_id_guid = get_guid_from_run_id(conn=conn,
+                                                   run_id=run_id)
+                guids.append(run_id_guid)
+            else:
+                raise RuntimeError(f'run id {run_id} and above do not'
+                                   ' exist in the database')
```

Comment:
What does above mean in this context. I think `run id {run_id} and above do not exist in the database` would be fine

Reply:
It is an extra (maybe not so necessary) explanation to say if for example run_id=3 doesn't exist, more than 3 also not available. 

you mean this is fine, right?:
run id {run_id} does not exist in the database

Reply:
Yes I think that's fine. Otherwise we could iterate through all of them to collect all the missing ones before raising but I don't think that is worth the effort.

```
missing = []
...
else:
    missing.append(run_id)
...
if len(missing) > 0
   raise RuntimeError(f"{missing} run_id(s) do not exist ...")
```

Reply:
Sure.  Also there might be a large database with a lot of runs that iterating over those is just wasting resources.